src/weapons.cpp

```diff
@@ -650,12 +660,13 @@ void WeaponDistance::configureWeapon(const ItemType& it)
 	Weapon::configureWeapon(it);
 }
 
-bool WeaponDistance::useWeapon(Player* player, Item* item, Creature* target) const
+bool WeaponDistance::useWeapon(const std::shared_ptr<Player>& player, const std::shared_ptr<Item>& item,
+                               const std::shared_ptr<Creature>& target) const
 {
 	int32_t damageModifier = 0;
 	const ItemType& it = Item::items[id];
 	if (it.weaponType == WEAPON_AMMO) {
-		Item* mainWeaponItem = player->getWeapon(true);
+		auto mainWeaponItem = player->getWeapon(true);
```

Comment:
We can use `const auto&` and avoid this hard-to-read
```suggestion
		if (const auto& mainWeaponItem = player->getWeapon(true)) {
			if (const Weapon* mainWeapon = g_weapons->getWeapon(mainWeaponItem)) {
				damageModifier = mainWeapon->playerWeaponCheck(player, target, mainWeaponItem->getShootRange());
			} else {
				damageModifier = playerWeaponCheck(player, target, mainWeaponItem->getShootRange());
			}
		}
```